### PR TITLE
[Serverless Mini Agent] Add Default Values for peer_tags and is_trace_root when Generating Trace Metrics

### PR DESCRIPTION
**Pull request by duncanpharvey**
# What does this PR do?

Adds `#[serde(default)]` for `peer_tags` and `is_trace_root` to prevent errors when stats payloads do not include these fields.

# Motivation

https://datadoghq.atlassian.net/browse/SVLS-4933

Deserialization errors when sending trace stats from Azure Functions and Google Cloud Functions.

```
2024-06-17T17:56:38Z   [Information]   [2024-06-17T17:56:37Z ERROR datadog_trace_mini_agent::http_utils] Error deserializing trace stats from request body: Error deserializing stats from request body: missing field `PeerTags`
```

```
2024-06-17T18:58:42Z   [Information]   [2024-06-17T18:58:42Z ERROR datadog_trace_mini_agent::http_utils] Error deserializing trace stats from request body: Error deserializing stats from request body: missing field `IsTraceRoot`
```

# Additional Notes

# How to test the change?

- Build mini agent locally
- Deploy Azure Function or Google Cloud Function with mini agent binary in root path
- Set `DD_MINI_AGENT_PATH`
  - Azure: `/home/site/wwwroot/datadog-serverless-trace-mini-agent`
  - Google: `/workspace/datadog-serverless-trace-mini-agent`

### CODE REVIEW

**Review by astuyve — APPROVED** [review]

**Review by bantonsson — COMMENTED** [review]
Could we add tests for this?

**Comment by duncanpharvey** [comment]
@bantonsson Added unit tests as requested!

**Review by bantonsson — APPROVED** [review]

File: trace-protobuf/Cargo.toml [hunk 1]
```diff
@@ -10,10 +10,16 @@ license.workspace = true
 prost = "0.11.6"
 serde = { version = "1.0.145", features = ["derive"] }
 serde_bytes = "0.11.9"
+serde_json = "1.0"
+tokio = { version = "1", features = ["macros", "rt-multi-thread"] }
```
[the remark this answers is not in the document]
Comment (duncanpharvey) [reply trace-protobuf/Cargo.toml]: Yep, fixed

File: trace-protobuf/src/pb.rs [hunk 2]
```diff
@@ -429,9 +429,11 @@ pub struct ClientGroupedStats {
     /// peer_tags are supplementary tags that further describe a peer entity
     /// E.g., `grpc.target` to describe the name of a gRPC peer, or `db.hostname` to describe the name of peer DB
     #[prost(string, repeated, tag = "16")]
+    #[serde(default)]
```
[the remark this answers is not in the document]
Comment (duncanpharvey) [reply trace-protobuf/src/pb.rs:432]: Done, let me know if this is what you were thinking

https://github.com/DataDog/libdatadog/blob/e021eed88d20fef9bc19e8a207f8b2aad33e9f7e/trace-protobuf/build.rs#L93-L94

[the remark this answers is not in the document]
Comment (bantonsson) [reply trace-protobuf/src/pb.rs:432]: That should do it.

**Pull request merged**